build-release-tools/lib/ManifestGenerator.py

```diff
@@ -119,9 +121,25 @@ def generate_manifest(self):
 class SpecifyDayManifestGenerator(ManifestGenerator):
     def __init__(self, dest, branch, date, builddir, git_credential=None, force=False, jobs=1):
         self._date = date
+        self._author = config.gitbit_identity["username"]
         super(SpecifyDayManifestGenerator, self).__init__(dest, branch, builddir, git_credential=git_credential, force=force, jobs=jobs)
 
     def update_repositories_commit(self, repositories):
         for repo in repositories:
             repo_dir = self.directory_for_repo(repo)
-            repo["commit-id"] = self.repo_operator.get_lastest_merge_commit_before_date(repo_dir,self._date)
+            merge_commit = None
+            jenkins_commit = None
+            try:
+                # Get the last merge commit before the date
+                merge_commit = self.repo_operator.get_latest_merge_commit_before_date(repo_dir,self._date)
+                # Get the last commit of Jenkins before the date
+                jenkins_commit = self.repo_operator.get_latest_author_commit_before_date(repo_dir,self._date, self._author)
```

Comment:
the function calling is not fixed to Jenkins, and the 'self._auther' variable is from config. so it's better get commit firstly, then compare with JenkinsRHD user.